### PR TITLE
Refactor getting devops operator

### LINKED ISSUE #38: Wrong parse for bearer token
Recently, I was using Postman to request ks-devops apis and `Unauthorized` error returned. 
I found a wrong parse for bearer token, which should start with `Bearer` instead of `bearer`:

https://github.com/LinuxSuRen/ks-devops/blob/3d29680f8b368d95e3a22838c45cf419164da9cd/pkg/apiserver/filters/requestinfo.go#L79

Please have a look at [The OAuth 2.0 Authorization Framework: Bearer Token Usage](https://datatracker.ietf.org/doc/html/rfc6750#section-2.1) as well.

### PR DESCRIPTION
**Pull request by JohnNiang**
Due to change of devops authentication, and devops operator has no ability to authenticate request, so we have to intercept request before invoking methods of devops operator. This PR tries to fix alph3 api handler.

By the way, this PR has merged another PR: https://github.com/LinuxSuRen/ks-devops/pull/29.

Also fix #38

### CODE REVIEW

File: pkg/kapis/devops/v1alpha3/handler.go [hunk 1]
```diff
@@ -248,95 +260,89 @@ func (h *devopsHandler) UpdatePipeline(request *restful.Request, response *restf
...
 func (h *devopsHandler) ListCredential(request *restful.Request, response *restful.Response) {
 	devops := request.PathParameter("devops")
 	query := query.ParseQueryParameter(request)
-	objs, err := h.devops.ListCredentialObj(devops, query)
 
-	if err != nil {
-		klog.Error(err)
-		if errors.IsNotFound(err) {
-			api.HandleNotFound(response, request, err)
+	if client, err := h.getDevOps(request); err == nil {
```
Comment (LinuxSuRen) [inline pkg/kapis/devops/v1alpha3/handler.go:331]: It does not work well. See the following screenshot.

image

image

image

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

File: pkg/kapis/devops/v1alpha3/handler.go (hunk 1 above)
Comment (JohnNiang) [reply pkg/kapis/devops/v1alpha3/handler.go:331]: I'm sorry for that I don't test most apis manually but pipeline list api. And I also discarded pr https://github.com/LinuxSuRen/ks-devops/pull/29 by mistake.

Comment (JohnNiang) [reply pkg/kapis/devops/v1alpha3/handler.go:331]: Thanks for your patient review and test.

Comment (LinuxSuRen) [reply pkg/kapis/devops/v1alpha3/handler.go:325]: > I'm sorry for that I don't test most apis manually but pipeline list API

I did that work from https://github.com/LinuxSuRen/ks-devops/pull/21/

Please make sure that all changes was tested before others start to review your changes.

**Comment by JohnNiang** [comment]
Hi @LinuxSuRen , I have tested all apis which I've changed. Please see as follow:

image

Meanwhile, I've exclude which lint failed files in .golangci.yaml.

File: pkg/apiserver/filters/requestinfo.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ func WithRequestInfo(handler http.Handler, resolver request.RequestInfoResolver)
 
 func injectToken(req *http.Request) (requestWithTokenContext *http.Request) {
 	authorization := req.Header.Get("Authorization")
-	token := strings.ReplaceAll(authorization, "bearer ", "")
+	token := strings.ReplaceAll(authorization, "Bearer ", "")
```
Comment (LinuxSuRen) [inline pkg/apiserver/filters/requestinfo.go:79]: This is excellent. But the README file is still incorrect. See also https://github.com/kubesphere-sigs/ks-devops#apis

File: .golangci.yml [hunk 3]
```diff
@@ -1,4 +1,103 @@
...
+
+  skip-dirs:
+    - .git
+    - .github
+    - bin
+    - charts
+    - docs
+    - hack
+  skip-files:
+    - cmd/allinone/app/entrypoint.go
```
Comment (JohnNiang) [inline .golangci.yml:18]: Exclude those lint failed files. If anyone confirm that any files are lint-well, you can remove corresponding lines.

File: pkg/apiserver/filters/requestinfo.go (hunk 2 above)
Comment (JohnNiang) [reply pkg/apiserver/filters/requestinfo.go:79]: Good catch. I'll rectify it soon.

Comment (JohnNiang) [reply pkg/apiserver/filters/requestinfo.go:79]: Done.

**Review by LinuxSuRen — APPROVED** [review]
LGTM

**Pull request merged**